spec/rubocop/cop/rspec/change_by_zero_spec.rb

```diff
@@ -25,6 +25,56 @@
     RUBY
   end
 
+  it 'registers an offense when using `a_block_changing` ' \
+     'and argument to `by` is zero' do
+    expect_offense(<<-RUBY)
+      it do
+        expect { foo }.to a_block_changing(Foo, :bar).by(0)
+                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Prefer `not_to a_block_changing` over `to a_block_changing.by(0)`.
+        expect { foo }.to a_block_changing(::Foo, :bar).by(0)
+                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Prefer `not_to a_block_changing` over `to a_block_changing.by(0)`.
+        expect { foo }.to a_block_changing { Foo.bar }.by(0)
+                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Prefer `not_to a_block_changing` over `to a_block_changing.by(0)`.
+        expect { foo }.to a_block_changing(Foo, :bar).by 0
+                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Prefer `not_to a_block_changing` over `to a_block_changing.by(0)`.
```

Comment:
Does it make sense to cover all of that?

Reply:
`a_block_changing` is just an alias of `change` and `changing`, right? I am not sure in which context it is meant to be used. Grammatically, I think only `change` could be used in this situation
```
expect { foo }.not_to change(Foo, :bar)
```

`a_block_changing` was added to RSpec in https://github.com/rspec/rspec-expectations/commit/9cf0fea4100ba347f75da014046e628edfac5de2, but it is unclear what it is used for. Perhaps composable matchers?

Reply:
Are you hinting that `a_block_changing` is not used in practice, @bquorning ? I can confirm that by scanning real-world-rspec, the only usages I could find are from rspec-expectations. It's hard to say if `changing` is used or not, as it's a commonly used wording.

Reply:
Not hinting anything, except that I have never used it myself 😅

Reply:
We have changed the autocorrect to the following format. WDYT?

```ruby
expect { run }.not_to change(Foo, :bar)
```